src/serac/physics/base_physics.hpp

```diff
@@ -107,6 +107,14 @@ class BasePhysics {
    */
   virtual std::vector<double> timesteps() const;
 
+  /**
+   * @brief Base method to reset physics states to zero.  This does not reset design parameters or shape.
```

Comment:
I'm OK with this as it is documented. Any preference @barreracruz @kswartz92 ?

Reply:
I like having a reset method that does not change parameters/shape displacement, i.e., just prepares the solver for a new transient evaluation and sensitivity calculation

Reply:
also, naming thoughts for everyone?  I considered initialize, reset, reinitialize, clear, and zeroState.

Reply:
I like `initialize` because we use it both for the first and subsequent calls, but I'd also be OK with `reset`.

Reply:
If this is not resetting/initializing everything. I think it would be more understandable if it had something in the name to clarify that. Would having `state` in the name be enough? `clearEverythingButShapeAndParameters` is probably too verbose =)

I am not sure I personally like `initialize` because it seems weird to do this:

```
initialize()
// do lots of stuff
initialize()
// do more stuff
```

Would the user be calling initialize in the first case or is it hidden in a constructor?

But none of these are bad..

Reply:
I changes the public api on base_physics to be resetStates().  I think this makes more sense for external usage, as Chris mentions.  For the internal non-virtual methods, I kept as initialize because that is how they are used internally.